### PR TITLE
Redirects to login page if user not logged in at welcome page

### PR DESCRIPTION
**Pull request by vera-liu**
Issue: [https://github.com/airbnb/superset/issues/1703](url)

giphy 16

@ascott

### CODE REVIEW

File: superset/views.py [hunk 1]
```diff
@@ -2552,6 +2552,8 @@ def show_traceback(self):
     @expose("/welcome")
     def welcome(self):
         """Personalized welcome page"""
+        if not g.user.get_id():
+            return redirect(appbuilder.get_url_for_login)
```
Comment (bkyryliuk) [inline superset/views.py]: is get_url_for_login a function or attr ?

Comment (vera-liu) [inline superset/views.py]: it's a property of appbuilder

File: superset/views.py [hunk 2]
```diff
@@ -2552,6 +2552,8 @@ def show_traceback(self):
     @expose("/welcome")
     def welcome(self):
         """Personalized welcome page"""
+        if not g.user or not g.user.get_id():
```
Comment (ascott) [inline superset/views.py]: should this be `and` rather than `or`?

**Comment by ascott** [comment]
lgtm

**Comment by bkyryliuk** [comment]
LGTM once tested

File: superset/views.py (hunk 1 above)
Comment (Gowtham95india) [inline superset/views.py]: Why making it complicated?

@has_access will do the trick.

**Pull request merged**